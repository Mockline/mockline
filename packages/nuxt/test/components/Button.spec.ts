import { describe, it, expect } from 'vitest'
import ComponentRender from '../component-render'
import { MButton } from '#components'

describe('Button', () => {
  it.each([
    ['basic case', { }],
    ['leading icon', { props: { leading: true, icon: 'heroicons-check' } }],
    ['black solid', { props: { color: 'black', variant: 'solid' } }],
    ['rounded full', { props: { ui: { rounded: 'rounded-full' } } }],
    ['<UButton icon="i-heroicons-pencil-square" size="sm" color="primary" square variant="solid" />']
    // @ts-expect-error
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof UButton.props>) => {
    if (options !== undefined) {
      options.slots = options.slots || { default: () => 'label' }
      options.slots.default = options.slots.default || (() => 'label')
    }
    const html = await ComponentRender(nameOrHtml, options, MButton)
    expect(html).toMatchSnapshot()
  })
})
