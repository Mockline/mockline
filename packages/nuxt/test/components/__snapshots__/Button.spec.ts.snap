// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > can mount component 1`] = `
"<div id="test-wrapper"><button class="relative inline-flex items-center justify-center font-medium cursor-pointer select-none focus-visible:outline-none touch-none tap-highlight-none rounded-[var(--radius-base)] text-sm gap-1.5 text-white bg-[var(--color-accent)] hover:bg-[var(--color-accent-hover)] focus-visible:ring-2 focus-visible:ring-[var(--color-accent)] focus-visible:ring-offset-2 disabled:bg-[var(--color-accent)]/75 shadow-sm p-1.5 transition-colors duration-200 ease-in-out active:translate-y-px">
    <!--v-if-->
    <!--v-if-->
    <!--v-if-->
  </button></div>"
`;

exports[`Button > can mount some component 1`] = `
"<div id="test-wrapper"><button class="relative inline-flex items-center justify-center font-medium cursor-pointer select-none focus-visible:outline-none touch-none tap-highlight-none rounded-[var(--radius-base)] text-sm gap-1.5 text-white bg-[var(--color-accent)] hover:bg-[var(--color-accent-hover)] focus-visible:ring-2 focus-visible:ring-[var(--color-accent)] focus-visible:ring-offset-2 disabled:bg-[var(--color-accent)]/75 shadow-sm p-1.5 transition-colors duration-200 ease-in-out active:translate-y-px">
    <!--v-if-->
    <!--v-if-->
    <!--v-if-->
  </button></div>"
`;

exports[`Button > renders <UButton icon="i-heroicons-pencil-square" size="sm" color="primary" square variant="solid" /> correctly 1`] = `
"<button class="relative inline-flex items-center justify-center font-medium cursor-pointer select-none focus-visible:outline-none touch-none tap-highlight-none rounded-[var(--radius-base)] text-xs gap-1.5 text-white bg-[var(--color-accent)] hover:bg-[var(--color-accent-hover)] focus-visible:ring-2 focus-visible:ring-[var(--color-accent)] focus-visible:ring-offset-2 disabled:bg-[var(--color-accent)]/75 shadow-sm p-1.5 transition-colors duration-200 ease-in-out active:translate-y-px"><span class="iconify i-heroicons:pencil-square shrink-0 relative size-4" aria-hidden="true"></span>
  <!--v-if-->
  <!--v-if-->
</button>"
`;

exports[`Button > renders basic case correctly 1`] = `
"<button class="relative inline-flex items-center justify-center font-medium cursor-pointer select-none focus-visible:outline-none touch-none tap-highlight-none rounded-[var(--radius-base)] px-2.5 py-1 text-sm gap-1.5 text-white bg-[var(--color-accent)] hover:bg-[var(--color-accent-hover)] focus-visible:ring-2 focus-visible:ring-[var(--color-accent)] focus-visible:ring-offset-2 disabled:bg-[var(--color-accent)]/75 shadow-sm transition-colors duration-200 ease-in-out active:translate-y-px">
  <!--v-if--><span class="truncate relative">label</span>
  <!--v-if-->
</button>"
`;

exports[`Button > renders black solid correctly 1`] = `
"<button class="relative inline-flex items-center justify-center font-medium cursor-pointer select-none focus-visible:outline-none touch-none tap-highlight-none rounded-[var(--radius-base)] px-2.5 py-1 text-sm gap-1.5 transition-colors duration-200 ease-in-out active:translate-y-px">
  <!--v-if--><span class="truncate relative">label</span>
  <!--v-if-->
</button>"
`;

exports[`Button > renders leading icon correctly 1`] = `
"<button class="relative inline-flex items-center justify-center font-medium cursor-pointer select-none focus-visible:outline-none touch-none tap-highlight-none rounded-[var(--radius-base)] px-2.5 py-1 text-sm gap-1.5 text-white bg-[var(--color-accent)] hover:bg-[var(--color-accent-hover)] focus-visible:ring-2 focus-visible:ring-[var(--color-accent)] focus-visible:ring-offset-2 disabled:bg-[var(--color-accent)]/75 shadow-sm transition-colors duration-200 ease-in-out active:translate-y-px"><span class="iconify i-heroicons:check shrink-0 relative size-4" aria-hidden="true"></span><span class="truncate relative">label</span>
  <!--v-if-->
</button>"
`;

exports[`Button > renders rounded full correctly 1`] = `
"<button class="relative inline-flex items-center justify-center font-medium cursor-pointer select-none focus-visible:outline-none touch-none tap-highlight-none rounded-[var(--radius-base)] px-2.5 py-1 text-sm gap-1.5 text-white bg-[var(--color-accent)] hover:bg-[var(--color-accent-hover)] focus-visible:ring-2 focus-visible:ring-[var(--color-accent)] focus-visible:ring-offset-2 disabled:bg-[var(--color-accent)]/75 shadow-sm transition-colors duration-200 ease-in-out active:translate-y-px" ui="[object Object]">
  <!--v-if--><span class="truncate relative">label</span>
  <!--v-if-->
</button>"
`;
